internal/sign/rsa_test.go

```diff
@@ -49,3 +60,51 @@ func TestInvalidHash(t *testing.T) {
 	_, err := RSASignSHA1Digest(invalidDigest, "testdata/rsa.priv", "hunter2")
 	require.EqualError(t, err, "digest is not a SHA1 hash")
 }
+
+func TestRSAVerifySHA1DigestError(t *testing.T) {
+	rsaKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		t.Fatal(err)
+	}
+	asn1Bytes, err := asn1.Marshal(rsaKey.PublicKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ecdsaKey, err := ecdsa.GenerateKey(elliptic.P521(), rand.Reader)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	x509Bytes, err := x509.MarshalPKIXPublicKey(&ecdsaKey.PublicKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pemKeys := []*pem.Block{
+		{
+			Type:  "PUBLIC KEY",
+			Bytes: asn1Bytes,
+		},
+		{
+			Type:  "PUBLIC KEY",
+			Bytes: x509Bytes,
+		},
+	}
+
+	for _, pemKey := range pemKeys {
+		cwd, _ := os.Getwd()
+
+		keyFile, err := ioutil.TempFile(cwd, "*.pem")
```

Comment:
Maybe just create the temp file in the OS temp folder (`ioutil.TempFile("", "*.pem")`). This way the working tree is not polluted if the test is interrupted by CTRL+C in the wrong time.

Reply:
fixed!